Lib/defcon/objects/font.py

```diff
@@ -1382,11 +1408,92 @@ def _convertToFormatVersion1RoboFabData(self, libCopy):
                         finalValues.append([])
                     finalValues[-1].append(value)
                 hintData[libKey] = finalValues
-        for key, value in hintData.items():
+        for key, value in list(hintData.items()):
             if value is None:
                 del hintData[key]
         libCopy["org.robofab.postScriptHintData"] = hintData
 
+    # -----------------------------
+    # Serialization/Deserialization
+    # -----------------------------
+
+    def getDataForSerialization(self, **kwargs):
+        from functools import partial
+
+        simple_get = partial(getattr, self)
+        serialize = lambda item: item.getDataForSerialization()
+        serialized_get = lambda key: serialize(simple_get(key))
+
+        getters = (
```

Comment:
I agree, a good way to extend the serialization api in subclasses would be the right thing to do. When I was writing that code it was more like an urgent thing that needed to happen, so that Trufont would get released with usable undo/redo. It was in a way self contained and had this small API footprint. I don't remember my original reasoning for that design choice, though. A good next step would be to make it better for subclassing and expose more of its guts to the programmer.